openedx/features/edly/migrations/0012_auto_20230418_1310.py

```diff
@@ -0,0 +1,40 @@
+# Generated by Django 2.2.16 on 2023-04-18 13:10
+
+from django.db import migrations
+
+
+def migrate_user(apps, schema_editor):
+    EdlyUserProfile = apps.get_model('edly', 'EdlyUserProfile')
+    EdlySubOrganization = apps.get_model('edly', 'EdlySubOrganization')
+    EdlyMultiSiteAccess = apps.get_model('edly', 'EdlyMultiSiteAccess')
+    for user in EdlyUserProfile.objects.all():
+        django_user = user.user
+        groups = django_user.groups.all()
+        for edly_sub_org in user.edly_sub_organizations.all():
+            edly_org = edly_sub_org.edly_organization
+            sub_orgs = [edly_sub_org] if not edly_org.enable_all_edly_sub_org_login else EdlySubOrganization.objects.filter(
+                edly_organization=edly_org
+            )
+            for sub_org in sub_orgs:
+                edly_access_user, _ = EdlyMultiSiteAccess.objects.get_or_create(
+                    sub_org=sub_org,
+                    user=django_user
+                )
+                edly_access_user.groups.add(*groups)
+                edly_access_user.is_blocked = user.is_blocked
+                edly_access_user.save()
+
+
+def backward_migration(apps, schema_editor):
+    EdlyMultiSiteAccess = apps.get_model('edly', 'EdlyMultiSiteAccess')
+    EdlyMultiSiteAccess.objects.all().delete()
```

Comment:
will the `backward_migration` run each time?

Reply:
nope, it only runs when we we explicitly run backward migration
`python manage.py migrate app_name 0000`